src/ElementQueries.js

```diff
@@ -350,25 +351,36 @@
             }
             if ('string' === typeof rules) {
                 rules = rules.toLowerCase();
-                if (-1 !== rules.indexOf('min-width') || -1 !== rules.indexOf('max-width')) {
+                if (isElementQuery(rules)) {
                     extractQuery(rules);
                 }
             } else {
-                for (var i = 0, j = rules.length; i < j; i++) {
-                    if (1 === rules[i].type) {
-                        selector = rules[i].selectorText || rules[i].cssText;
-                        if (-1 !== selector.indexOf('min-height') || -1 !== selector.indexOf('max-height')) {
-                            extractQuery(selector);
-                        }else if(-1 !== selector.indexOf('min-width') || -1 !== selector.indexOf('max-width')) {
-                            extractQuery(selector);
-                        }
-                    } else if (4 === rules[i].type) {
-                        readRules(rules[i].cssRules || rules[i].rules);
+                for (var i = 0, j = rules.length, r; i < j; i++) {
+                    r = rules[i];
+                    switch (r.type) {
+                      case 1:   // CSSRule.STYLE_RULE
+                          selector = r.selectorText || r.cssText;
+                          if (isElementQuery(selector)) {
+                              extractQuery(selector);
+                          }
+                          break;
+                      case 3:   // CSSRule.IMPORT_RULE
+                          r = r.styleSheet;  // then continue to next case statement
```

Comment:
They're almost equivalent. 

It is said that `switch/case` is more intuitive and easily optimized by JS engine. If count of conditions growing, `switch/case` usually runs faster than multiple `if-else-if`. 

But position of `case 3` and `case 4` must not be exchanged, since no `break` at the end of `case 3` which continuing to execute `case 4`.

@see http://archive.oreilly.com/pub/a/server-administration/excerpts/even-faster-websites/writing-efficient-javascript.html